### PR TITLE
Add the closest and average closest collision detectors

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/70891

This PR adds 2 new collision detectors we need to common:

* one to find unique closest matches
* one to find unique closest matches with a delta calculated on the average distance

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $b is never used | [Average_Proximity_Start_Detector.php:20](https://github.com/moderntribe/tribe-common/blob/16d4045f36db61a6c3e1dcafd969d971d2b74cab/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L20)
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Average_Proximity_Start_Detector.php:24](https://github.com/moderntribe/tribe-common/blob/16d4045f36db61a6c3e1dcafd969d971d2b74cab/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L24)
:warning: | The method parameter $a is never used | [Average_Proximity_Start_Detector.php:66](https://github.com/moderntribe/tribe-common/blob/16d4045f36db61a6c3e1dcafd969d971d2b74cab/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L66)
:warning: | The method parameter $b is never used | [Average_Proximity_Start_Detector.php:66](https://github.com/moderntribe/tribe-common/blob/16d4045f36db61a6c3e1dcafd969d971d2b74cab/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L66)
:warning: | The method parameter $segment is never used | [Closest_Unique_Start_Detector.php:23](https://github.com/moderntribe/tribe-common/blob/16d4045f36db61a6c3e1dcafd969d971d2b74cab/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L23)
:warning: | The method parameter $b_ends is never used | [Closest_Unique_Start_Detector.php:23](https://github.com/moderntribe/tribe-common/blob/16d4045f36db61a6c3e1dcafd969d971d2b74cab/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L23)
:warning: | The method parameter $report is never used | [Closest_Unique_Start_Detector.php:23](https://github.com/moderntribe/tribe-common/blob/16d4045f36db61a6c3e1dcafd969d971d2b74cab/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L23)
:warning: | The method parameter $a is never used | [Closest_Unique_Start_Detector.php:72](https://github.com/moderntribe/tribe-common/blob/16d4045f36db61a6c3e1dcafd969d971d2b74cab/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L72)
:warning: | The method parameter $b is never used | [Closest_Unique_Start_Detector.php:72](https://github.com/moderntribe/tribe-common/blob/16d4045f36db61a6c3e1dcafd969d971d2b74cab/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L72)
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Closest_Unique_Start_Detector.php:76](https://github.com/moderntribe/tribe-common/blob/16d4045f36db61a6c3e1dcafd969d971d2b74cab/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L76)
:warning: | The method parameter $a is never used | [Closest_Unique_Start_Detector.php:117](https://github.com/moderntribe/tribe-common/blob/16d4045f36db61a6c3e1dcafd969d971d2b74cab/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L117)
:warning: | The method parameter $b is never used | [Closest_Unique_Start_Detector.php:117](https://github.com/moderntribe/tribe-common/blob/16d4045f36db61a6c3e1dcafd969d971d2b74cab/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L117)
:x: | Blank line found after control structure | [Closest_Unique_Start_Detector.php:188](https://github.com/moderntribe/tribe-common/blob/16d4045f36db61a6c3e1dcafd969d971d2b74cab/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L188)
:warning: | The method parameter $a is never used | [Closest_Unique_Start_Detector.php:221](https://github.com/moderntribe/tribe-common/blob/16d4045f36db61a6c3e1dcafd969d971d2b74cab/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L221)
:warning: | The method parameter $b is never used | [Closest_Unique_Start_Detector.php:221](https://github.com/moderntribe/tribe-common/blob/16d4045f36db61a6c3e1dcafd969d971d2b74cab/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L221)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $b is never used | [Average_Proximity_Start_Detector.php:20](https://github.com/moderntribe/tribe-common/blob/60d539e911195f975f0fd76a08c8aae2273e759b/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L20)
:warning: | The method parameter $a is never used | [Average_Proximity_Start_Detector.php:67](https://github.com/moderntribe/tribe-common/blob/60d539e911195f975f0fd76a08c8aae2273e759b/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L67)
:warning: | The method parameter $b is never used | [Average_Proximity_Start_Detector.php:67](https://github.com/moderntribe/tribe-common/blob/60d539e911195f975f0fd76a08c8aae2273e759b/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L67)
:warning: | The method parameter $segment is never used | [Closest_Unique_Start_Detector.php:23](https://github.com/moderntribe/tribe-common/blob/60d539e911195f975f0fd76a08c8aae2273e759b/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L23)
:warning: | The method parameter $b_ends is never used | [Closest_Unique_Start_Detector.php:23](https://github.com/moderntribe/tribe-common/blob/60d539e911195f975f0fd76a08c8aae2273e759b/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L23)
:warning: | The method parameter $report is never used | [Closest_Unique_Start_Detector.php:23](https://github.com/moderntribe/tribe-common/blob/60d539e911195f975f0fd76a08c8aae2273e759b/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L23)
:warning: | The method parameter $a is never used | [Closest_Unique_Start_Detector.php:72](https://github.com/moderntribe/tribe-common/blob/60d539e911195f975f0fd76a08c8aae2273e759b/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L72)
:warning: | The method parameter $b is never used | [Closest_Unique_Start_Detector.php:72](https://github.com/moderntribe/tribe-common/blob/60d539e911195f975f0fd76a08c8aae2273e759b/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L72)
:warning: | The method parameter $a is never used | [Closest_Unique_Start_Detector.php:118](https://github.com/moderntribe/tribe-common/blob/60d539e911195f975f0fd76a08c8aae2273e759b/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L118)
:warning: | The method parameter $b is never used | [Closest_Unique_Start_Detector.php:118](https://github.com/moderntribe/tribe-common/blob/60d539e911195f975f0fd76a08c8aae2273e759b/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L118)
:warning: | The method parameter $a is never used | [Closest_Unique_Start_Detector.php:221](https://github.com/moderntribe/tribe-common/blob/60d539e911195f975f0fd76a08c8aae2273e759b/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L221)
:warning: | The method parameter $b is never used | [Closest_Unique_Start_Detector.php:221](https://github.com/moderntribe/tribe-common/blob/60d539e911195f975f0fd76a08c8aae2273e759b/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L221)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | There must be a single space before the last parenthesis | [Average_Proximity_Start_Detector.php:18](https://github.com/moderntribe/tribe-common/blob/84b183b999002d0b781271145980f531c9865fc5/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L18)
:x: | There must be a single space after the first parenthesis | [Average_Proximity_Start_Detector.php:18](https://github.com/moderntribe/tribe-common/blob/84b183b999002d0b781271145980f531c9865fc5/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L18)
:x: | There must be a single space after the first parenthesis | [Average_Proximity_Start_Detector.php:18](https://github.com/moderntribe/tribe-common/blob/84b183b999002d0b781271145980f531c9865fc5/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L18)
:x: | There must be a single space before the last parenthesis | [Average_Proximity_Start_Detector.php:18](https://github.com/moderntribe/tribe-common/blob/84b183b999002d0b781271145980f531c9865fc5/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L18)
:warning: | The method parameter $b is never used | [Average_Proximity_Start_Detector.php:36](https://github.com/moderntribe/tribe-common/blob/84b183b999002d0b781271145980f531c9865fc5/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L36)
:warning: | The method parameter $a is never used | [Average_Proximity_Start_Detector.php:79](https://github.com/moderntribe/tribe-common/blob/84b183b999002d0b781271145980f531c9865fc5/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L79)
:warning: | The method parameter $b is never used | [Average_Proximity_Start_Detector.php:79](https://github.com/moderntribe/tribe-common/blob/84b183b999002d0b781271145980f531c9865fc5/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L79)
:x: | There must be a single space before the last parenthesis | [Average_Proximity_Start_Detector.php:121](https://github.com/moderntribe/tribe-common/blob/84b183b999002d0b781271145980f531c9865fc5/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L121)
:x: | There must be a single space after the first parenthesis | [Average_Proximity_Start_Detector.php:121](https://github.com/moderntribe/tribe-common/blob/84b183b999002d0b781271145980f531c9865fc5/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L121)
:x: | There must be a single space after the first parenthesis | [Average_Proximity_Start_Detector.php:121](https://github.com/moderntribe/tribe-common/blob/84b183b999002d0b781271145980f531c9865fc5/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L121)
:x: | There must be a single space before the last parenthesis | [Average_Proximity_Start_Detector.php:121](https://github.com/moderntribe/tribe-common/blob/84b183b999002d0b781271145980f531c9865fc5/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L121)
:warning: | The method parameter $a is never used | [Closest_Unique_Start_Detector.php:54](https://github.com/moderntribe/tribe-common/blob/84b183b999002d0b781271145980f531c9865fc5/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L54)
:warning: | The method parameter $b is never used | [Closest_Unique_Start_Detector.php:54](https://github.com/moderntribe/tribe-common/blob/84b183b999002d0b781271145980f531c9865fc5/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L54)
:warning: | The method parameter $a is never used | [Closest_Unique_Start_Detector.php:87](https://github.com/moderntribe/tribe-common/blob/84b183b999002d0b781271145980f531c9865fc5/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L87)
:warning: | The method parameter $b is never used | [Closest_Unique_Start_Detector.php:87](https://github.com/moderntribe/tribe-common/blob/84b183b999002d0b781271145980f531c9865fc5/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L87)
:warning: | The method parameter $a is never used | [Closest_Unique_Start_Detector.php:190](https://github.com/moderntribe/tribe-common/blob/84b183b999002d0b781271145980f531c9865fc5/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L190)
:warning: | The method parameter $b is never used | [Closest_Unique_Start_Detector.php:190](https://github.com/moderntribe/tribe-common/blob/84b183b999002d0b781271145980f531c9865fc5/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L190)
:warning: | The method parameter $segment is never used | [Closest_Unique_Start_Detector.php:204](https://github.com/moderntribe/tribe-common/blob/84b183b999002d0b781271145980f531c9865fc5/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L204)
:warning: | The method parameter $b_ends is never used | [Closest_Unique_Start_Detector.php:204](https://github.com/moderntribe/tribe-common/blob/84b183b999002d0b781271145980f531c9865fc5/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L204)
:warning: | The method parameter $report is never used | [Closest_Unique_Start_Detector.php:204](https://github.com/moderntribe/tribe-common/blob/84b183b999002d0b781271145980f531c9865fc5/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L204)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $b is never used | [Average_Proximity_Start_Detector.php:36](https://github.com/moderntribe/tribe-common/blob/bd786b7a9f8eea73e689b811865df85accae8e3b/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L36)
:warning: | The method parameter $a is never used | [Average_Proximity_Start_Detector.php:79](https://github.com/moderntribe/tribe-common/blob/bd786b7a9f8eea73e689b811865df85accae8e3b/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L79)
:warning: | The method parameter $b is never used | [Average_Proximity_Start_Detector.php:79](https://github.com/moderntribe/tribe-common/blob/bd786b7a9f8eea73e689b811865df85accae8e3b/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L79)
:warning: | The method parameter $b is never used | [Closest_Unique_Start_Detector.php:54](https://github.com/moderntribe/tribe-common/blob/bd786b7a9f8eea73e689b811865df85accae8e3b/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L54)
:warning: | The method parameter $b is never used | [Closest_Unique_Start_Detector.php:91](https://github.com/moderntribe/tribe-common/blob/bd786b7a9f8eea73e689b811865df85accae8e3b/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L91)
:warning: | The method parameter $b is never used | [Closest_Unique_Start_Detector.php:202](https://github.com/moderntribe/tribe-common/blob/bd786b7a9f8eea73e689b811865df85accae8e3b/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L202)
:warning: | The method parameter $segment is never used | [Closest_Unique_Start_Detector.php:220](https://github.com/moderntribe/tribe-common/blob/bd786b7a9f8eea73e689b811865df85accae8e3b/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L220)
:warning: | The method parameter $b_ends is never used | [Closest_Unique_Start_Detector.php:220](https://github.com/moderntribe/tribe-common/blob/bd786b7a9f8eea73e689b811865df85accae8e3b/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L220)
:warning: | The method parameter $report is never used | [Closest_Unique_Start_Detector.php:220](https://github.com/moderntribe/tribe-common/blob/bd786b7a9f8eea73e689b811865df85accae8e3b/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L220)
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Median_Proximity_Start_Detector.php:19](https://github.com/moderntribe/tribe-common/blob/bd786b7a9f8eea73e689b811865df85accae8e3b/src/Tribe/Collisions/Median_Proximity_Start_Detector.php#L19)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Collisions/Average_Proximity_Start_Detector.php [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+	 *               contains the segments that did collide with each colliding element of $a
+	 */
+	public function report_intersect( array $a, array $b ) {
+		$reported = call_user_func_array( array( 'parent', 'report_intersect' ), func_get_args() );
+
+		if ( empty( $reported ) ) {
+			return array();
+		}
+
+		$as = reset( $reported );
```
Comment (borkweb) [inline src/Tribe/Collisions/Average_Proximity_Start_Detector.php]: `$as` and `$bs` is a pretty cryptic set of names. Perhaps something that advertises what it holds a bit better without needing to refer to the comment above. Maybe `$a_dates`/`$b_dates` or `$first_date_pair`/`$last_date_pair`?

Rinse and repeat this comment on other strategies.

Comment (lucatume) [inline src/Tribe/Collisions/Average_Proximity_Start_Detector.php]: nope, we are in common here and we are comparing **segments** and not dates; dates happen to be segments but we could use the collision detection to find price ranges or anything that can be represented as a point or a segment.
I really want to keep this code free from knowing its  use and context  of use

Comment (borkweb) [inline src/Tribe/Collisions/Average_Proximity_Start_Detector.php]: That definitely makes sense! That being said, my comment still stands: `$as` and `$bs` are poor variable names.

Comment (lucatume) [inline src/Tribe/Collisions/Average_Proximity_Start_Detector.php]: `a_segments` and `b_segments`?
PHP `array_diff` uses `array1` and `array2` but that's not so much better... `segment_1` and `segment_2`?
My brain is jelly right now

Comment (barryhughes) [inline src/Tribe/Collisions/Average_Proximity_Start_Detector.php]: `$set_a` | `$set_b` ?

Comment (lucatume) [inline src/Tribe/Collisions/Average_Proximity_Start_Detector.php]: deal, thanks, will update

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $set_b is never used | [Average_Proximity_Start_Detector.php:37](https://github.com/moderntribe/tribe-common/blob/85a3ea0520d5acc019f4b404f2cfbefad0840c5d/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L37)
:warning: | The method parameter $set_a is never used | [Average_Proximity_Start_Detector.php:81](https://github.com/moderntribe/tribe-common/blob/85a3ea0520d5acc019f4b404f2cfbefad0840c5d/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L81)
:warning: | The method parameter $set_b is never used | [Average_Proximity_Start_Detector.php:81](https://github.com/moderntribe/tribe-common/blob/85a3ea0520d5acc019f4b404f2cfbefad0840c5d/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L81)
:warning: | The method parameter $b_set is never used | [Closest_Unique_Start_Detector.php:56](https://github.com/moderntribe/tribe-common/blob/85a3ea0520d5acc019f4b404f2cfbefad0840c5d/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L56)
:warning: | The method parameter $set_b is never used | [Closest_Unique_Start_Detector.php:94](https://github.com/moderntribe/tribe-common/blob/85a3ea0520d5acc019f4b404f2cfbefad0840c5d/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L94)
:warning: | The method parameter $set_b is never used | [Closest_Unique_Start_Detector.php:206](https://github.com/moderntribe/tribe-common/blob/85a3ea0520d5acc019f4b404f2cfbefad0840c5d/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L206)
:warning: | The method parameter $segment is never used | [Closest_Unique_Start_Detector.php:224](https://github.com/moderntribe/tribe-common/blob/85a3ea0520d5acc019f4b404f2cfbefad0840c5d/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L224)
:warning: | The method parameter $b_ends is never used | [Closest_Unique_Start_Detector.php:224](https://github.com/moderntribe/tribe-common/blob/85a3ea0520d5acc019f4b404f2cfbefad0840c5d/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L224)
:warning: | The method parameter $report is never used | [Closest_Unique_Start_Detector.php:224](https://github.com/moderntribe/tribe-common/blob/85a3ea0520d5acc019f4b404f2cfbefad0840c5d/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L224)
:warning: | The method parameter $set_b is never used | [Detection_Strategy.php:77](https://github.com/moderntribe/tribe-common/blob/85a3ea0520d5acc019f4b404f2cfbefad0840c5d/src/Tribe/Collisions/Detection_Strategy.php#L77)
:warning: | The method parameter $set_b is never used | [Detection_Strategy.php:183](https://github.com/moderntribe/tribe-common/blob/85a3ea0520d5acc019f4b404f2cfbefad0840c5d/src/Tribe/Collisions/Detection_Strategy.php#L183)
:warning: | The method parameter $b_set is never used | [Detection_Strategy.php:249](https://github.com/moderntribe/tribe-common/blob/85a3ea0520d5acc019f4b404f2cfbefad0840c5d/src/Tribe/Collisions/Detection_Strategy.php#L249)
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Median_Proximity_Start_Detector.php:19](https://github.com/moderntribe/tribe-common/blob/85a3ea0520d5acc019f4b404f2cfbefad0840c5d/src/Tribe/Collisions/Median_Proximity_Start_Detector.php#L19)

> via jenkins/codesniffer-phpcs

**Comment by lucatume** [comment]
Ok, I've renamed the vars, this should be good to go for another review

File: src/Tribe/Collisions/Closest_Unique_Start_Detector.php [hunk 2]
```diff
@@ -108,32 +111,33 @@ public function touch( array $a, array $b ) {
...
 	 *               contains the segments that did collide with each colliding element of $a
 	 */
-	public function report_intersect( array $a, array $b ) {
-		if ( empty( $a ) ) {
+	public function report_intersect( array $set_a, array $set_b ) {
+		if ( empty( $set_a ) ) {
 			return array();
 		}
 
 		$bs = func_get_args();
```
Comment (borkweb) [inline src/Tribe/Collisions/Closest_Unique_Start_Detector.php]: Change `$bs`?

File: src/Tribe/Collisions/Detection_Strategy.php [hunk 3]
```diff
@@ -14,24 +14,25 @@
...
+	 * @param array $set_b,... An array (ore more arrays) of elements each defining the start and end of a segment in
+	 *                         the format [<start>, <end>].
 	 *
 	 * @return array An array of arrays of elements each defining the start and end of a segment in the format
 	 *               [<start>, <end>]; the first array contains the segments of $a that collided while the second array
 	 *               contains the segments that did collide with each colliding element of $a
 	 */
-	public function report_touch( array $a, array $b ) {
+	public function report_touch( array $set_a, array $set_b ) {
 		$bs = func_get_args();
```
Comment (borkweb) [inline src/Tribe/Collisions/Detection_Strategy.php]: Change `$bs`?

File: src/Tribe/Collisions/Detection_Strategy.php [hunk 4]
```diff
@@ -63,56 +64,59 @@ public function report_touch( array $a, array $b ) {
...
+	 *                         segment
+	 *                         from the first array.
+	 * @param bool  $report    Whether colliding segments should be reported or not; defaults to `false`.
 	 *
 	 * @return array An array of elements each defining the start and end of a segment in the format [<start>, <end>].
 	 */
-	protected function collide( array $a, array $b, $discard = true, $report = false ) {
-		$bs = (array) $b;
+	protected function collide( array $set_a, array $set_b, $discard = true, $report = false ) {
+		$bs = (array) $set_b;
```
Comment (borkweb) [inline src/Tribe/Collisions/Detection_Strategy.php]: Change `$bs`?

File: src/Tribe/Collisions/Detection_Strategy.php [hunk 5]
```diff
@@ -194,25 +199,26 @@ public function diff( array $a, array $b ) {
...
+	 *                         [<start>,
+	 *                         <end>].
+	 * @param array $set_b,... An array (ore more arrays) of elements each defining the start and end of a segment in
+	 *                         the format [<start>, <end>].
 	 *
 	 * @return array An array of elements each defining the start and end of a segment in the format [<start>, <end>].
 	 */
-	public function touch( array $a, array $b ) {
+	public function touch( array $set_a, array $set_b ) {
 		$bs = func_get_args();
```
Comment (borkweb) [inline src/Tribe/Collisions/Detection_Strategy.php]: Change `$bs`?

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $set_b is never used | [Average_Proximity_Start_Detector.php:37](https://github.com/moderntribe/tribe-common/blob/0e4b673a5ba248cda81beb4641299423e9b345fd/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L37)
:warning: | The method parameter $set_a is never used | [Average_Proximity_Start_Detector.php:81](https://github.com/moderntribe/tribe-common/blob/0e4b673a5ba248cda81beb4641299423e9b345fd/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L81)
:warning: | The method parameter $set_b is never used | [Average_Proximity_Start_Detector.php:81](https://github.com/moderntribe/tribe-common/blob/0e4b673a5ba248cda81beb4641299423e9b345fd/src/Tribe/Collisions/Average_Proximity_Start_Detector.php#L81)
:warning: | The method parameter $b_set is never used | [Closest_Unique_Start_Detector.php:56](https://github.com/moderntribe/tribe-common/blob/0e4b673a5ba248cda81beb4641299423e9b345fd/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L56)
:warning: | The method parameter $set_b is never used | [Closest_Unique_Start_Detector.php:94](https://github.com/moderntribe/tribe-common/blob/0e4b673a5ba248cda81beb4641299423e9b345fd/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L94)
:warning: | The method parameter $set_b is never used | [Closest_Unique_Start_Detector.php:206](https://github.com/moderntribe/tribe-common/blob/0e4b673a5ba248cda81beb4641299423e9b345fd/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L206)
:warning: | The method parameter $segment is never used | [Closest_Unique_Start_Detector.php:224](https://github.com/moderntribe/tribe-common/blob/0e4b673a5ba248cda81beb4641299423e9b345fd/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L224)
:warning: | The method parameter $b_ends is never used | [Closest_Unique_Start_Detector.php:224](https://github.com/moderntribe/tribe-common/blob/0e4b673a5ba248cda81beb4641299423e9b345fd/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L224)
:warning: | The method parameter $report is never used | [Closest_Unique_Start_Detector.php:224](https://github.com/moderntribe/tribe-common/blob/0e4b673a5ba248cda81beb4641299423e9b345fd/src/Tribe/Collisions/Closest_Unique_Start_Detector.php#L224)
:warning: | The method parameter $set_b is never used | [Detection_Strategy.php:77](https://github.com/moderntribe/tribe-common/blob/0e4b673a5ba248cda81beb4641299423e9b345fd/src/Tribe/Collisions/Detection_Strategy.php#L77)
:warning: | The method parameter $set_b is never used | [Detection_Strategy.php:183](https://github.com/moderntribe/tribe-common/blob/0e4b673a5ba248cda81beb4641299423e9b345fd/src/Tribe/Collisions/Detection_Strategy.php#L183)
:warning: | The method parameter $b_set is never used | [Detection_Strategy.php:249](https://github.com/moderntribe/tribe-common/blob/0e4b673a5ba248cda81beb4641299423e9b345fd/src/Tribe/Collisions/Detection_Strategy.php#L249)
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Median_Proximity_Start_Detector.php:19](https://github.com/moderntribe/tribe-common/blob/0e4b673a5ba248cda81beb4641299423e9b345fd/src/Tribe/Collisions/Median_Proximity_Start_Detector.php#L19)

> via jenkins/codesniffer-phpcs

**Comment by borkweb** [comment]
♐️

**Pull request merged**